### ISSUE TITLE
automatic partition task fail with swap

### ISSUE THREAD

**Issue by towo2099** [calamares/calamares#2367]
> Hi! Thank you for helping improve Calamares. If you are seeing a problem in installing a specific distribution, you should **probably** report the problem in the distribution's bug tracker, first. That helps filter out issues with packaging, mis-configuration, etc. that Calamares has no control over. If you are a distribution packager or maintainer, this page is for you.

**Describe the bug**
If i choose automatic partitioning, e.g. erase disk, use swap "no hibernate", calamares install fails in creating partitions module.  Without swap or swapfile all is working as it should

**To Reproduce**
1. Start Calamares
2. Choose erase disk, with swap
3. Go fourther
4. Calamares Install will fail

**Expected behavior**
Install should be successfull.

**Screenshots and Logs**
```
Create a new partition (3.17 GiB, linuxswap) on ‘/dev/nvme0n1’


Job: Create new partition on device ‘/dev/nvme0n1’


Command: sfdisk --force --append /dev/nvme0n1

Failed to add partition ‘New Partition’ to device ‘/dev/nvme0n1’.

Failed to add partition ‘New Partition’ to device ‘/dev/nvme0n1’.

```
**Additional context**

Used Systems:  \

KDE neon User

```
TUXEDO OS (Ubuntu 22.04 with neon user packages)
Operating System: TUXEDO OS 3
KDE Plasma Version: 6.1.5
KDE Frameworks Version: 6.5.0
Gear Version: 24.08.0
Qt Version: 6.7.2
Kernel Version: 6.11.0-100005-tuxedo (64-bit)
Graphics Platform: X11
Processors: 6 × AMD Ryzen 7 2700X Eight-Core Processor
Memory: 7.7 GiB of RAM
Graphics Processor: virgl
Manufacturer: TUXEDO
Product Name: Tuxedo Qemu VM
```

**Comment by demmm** [comment]
Was Calamares rebuild on kpmcore 24.08.0?  What version of util-linux? (no answers, just trying to gather as much info as possible).

**Comment by towo2099** [comment]
> Was Calamares rebuild on kpmcore 24.08.0? What version of util-linux? (no answers, just trying to gather as much info as possible).

Yes, Calamares was rebuild against kpmcore 24.08.0 and util-linux is 2.37.2-4ubuntu3.4.

**Comment by dalto8** [comment]
Several people have noted that they are seeing this behavior when using kpmcore 24.08.0 and downgrading to the prior version resolves the issue.

**Comment by demmm** [comment]
Since everyone so far mentions sfdisk call is in the crash, has anyone tried compiling kpmcore, reverting the commit containing:
https://invent.kde.org/system/kpmcore/-/commit/067f1618513ff57e697a9bd521888435e399eeb0#b11aa0b697c9d50d6ab2bfa3cec04873d9d54b86_61_61 ?

**Comment by towo2099** [comment]
> Since everyone so far mentions sfdisk call is in the crash, has anyone tried compiling kpmcore, reverting the commit containing: https://invent.kde.org/system/kpmcore/-/commit/067f1618513ff57e697a9bd521888435e399eeb0#b11aa0b697c9d50d6ab2bfa3cec04873d9d54b86_61_61 ?

No, reverting does not help, even if i compile calamares against that version of kpmcore.

**Comment by killajoe** [comment]
> 
> No, reverting does not help, even if i compile calamares against that version of kpmcore.

Same for me with EndeavourOS.

But mentioning that Cachy and Arco can resolve issue without changing anything.. arco using git state to build calamares Cachy using not latest but the one before 3.0.8 ? 

EndeavourOS forking calamares and I can see we have still some parts from before 3.0.8 used.

**Comment by demmm** [comment]
KaOS is hit with the same issue.
Seems to be fixed (temp solution) by reverting:
https://invent.kde.org/system/kpmcore/-/commit/7f304be3e42b7efb72f990eacf2fa3d7640ffa23

So it looks like Calamares is looking for some of those removed options.

**Comment by killajoe** [comment]
Thank's @demmm tested this patch/revert works also on EndeavourOS
So now investigations can begun ;)

**Comment by demmm** [comment]
Tested on the same ISO if it can be replicated by just using kpmcore & partitionmanager.  Same disk setup used as in the Calamares test, no failing `sfdisk --force --append device` then:
![Screenshot_20240908_045909](https://github.com/user-attachments/assets/e9a493bf-6b35-4dd4-bd8d-3de92060574f)

**Comment by cruftex** [comment]
I was trying to do new install of KDE Neon from a fresh download of yesterday (neon-user-20240919-0714.iso) and hit the same errors as mentioned above. However, the error was the same without swap and encryption.

I install on a 4TB harddisk, and I could see that the installer creates a MBR partition table. When I run sfdisk manually I saw an error that GPT is needed for that disk size.

Then, in the BIOS I switched to "UEFI Only" boot and rebooted from the USB install disk. Now the installer is running smoothly!

**Comment by ezsit55** [comment]
Is there any movement on this issue? Is the only current solution to use version 3.3.8? If so, just let us know. Thanks.

**Comment by demmm** [comment]
As noted in this report, current solution is to revert one commit in kpmcore 24.08.  If you use an older kpmcore, you are not effected.

**Comment by ezsit55** [comment]
> As noted in this report, current solution is to revert one commit in kpmcore 24.08. If you use an older kpmcore, you are not effected.

That is not a long term solution. The kpmcore project does not show any sign of officially reverting their commit. Just because a Solus dev got Calamares to work with a modified build of kpmcore 24.08 does not really fix anything other than the immediate issue for for those willing to do the same.

**Comment by erikdubois** [comment]
Build of ArcoLinux on 18/10/24 resulted in Calamares not creating  a swap partition. Reading this post and following the advice of @demmm I rebuild only kpmcore and kept everything else untouched.

Use the pkgbuild from the official Arch Linux site and then added the patch from @demmm in it.

Here is the current PKGBUILD used on ArcoLinux

https://pastebin.com/QdxTvqw0

I will be testing that build the coming days. No more news then it is working for ArcoLinux.

**Comment by erikdubois** [comment]
After more than 20 installations with ext4, btrfs, xfs, jfs, reiserfs, f2fs, and varying sizes of swap, all installations were successful.

**Comment by jpetso** [comment]
I hit this issue on KDE neon, trying to install to a 953.87 GiB SSD, 1024209543168 bytes, 2000409264 sectors, sector size 512 bytes.

With `sudo sftrace -f --attach <calamares-pid> -s 10000 -v &> strace.log`, I found the exact start and end values that Calamares/kpmcore is passing to `sfdisk`. I can enter these into my own `sudo sfdisk --force --append /dev/nvme0n1` without Calamares/kpmcore swallowing its output. Relevant output:

```
Disk /dev/nvme0n1: 953.87 GiB, 1024209543168 bytes, 2000409264 sectors
Units: sectors of 1 * 512 = 512 bytes
(...)

Old situation:

Device          Start        End    Sectors   Size Type
/dev/nvme0n1p1   4096     618495     614400   300M EFI System
/dev/nvme0n1p2 618496 1929684292 1929065797 919.9G Linux filesystem

Type 'help' to get more information.

/dev/nvme0n1p3: start=1929684293 size=70724971
The last usable GPT sector is 2000409230, but 2000409263 is requested.
Failed to add #3 partition: Invalid argument
```

So, basically, Calamares/kpmcore is trying to create too large of a partition, exceeding the usable size of the drive (1929684293 + 70724971 = 2000409263).

How it ends up with this number, I haven't yet investigated. But it seems that some kind of space limitation isn't taken into account when calculating the requested size of the partition.

**Comment by jpetso** [comment]
Heya, I think I may have a fix for the size calculation, but I don't have a test setup at this point. Could someone in here give PR #2392 a shot, with kpmcore at the current version instead of the earlier reverted one? Thank you!
